File: src/main/java/io/dustin/finstage/financial/dto/CashFlowStatementResponse.java
package io.dustin.finstage.financial.dto;

import io.dustin.finstage.financial.domain.CashFlowStatement;
import lombok.Builder;
import lombok.Getter;

import java.time.LocalDate;

@Getter
@Builder
public class CashFlowStatementResponse {

    private final String symbol;
    private final LocalDate date;
    private final long operating_cash_flow;
    private final long investing_cash_flow;
    private final long financing_cash_flow;
    private final long free_cash_flow;

    public static CashFlowStatementResponse from(CashFlowStatement cashFlow) {
        return CashFlowStatementResponse.builder()
                .symbol(cashFlow.getSymbol())
                .date(cashFlow.getDate())
                .operating_cash_flow(cashFlow.getOperatingCashFlow())
                .investing_cash_flow(cashFlow.getInvestingCashFlow())
                .financing_cash_flow(cashFlow.getFinancingCashFlow())
                .free_cash_flow(cashFlow.getFreeCashFlow())
                .build();
    }
}

File: src/main/kotlin/io/dustin/finstage/financial/dto/CashFlowStatementResponse.kt
package io.dustin.finstage.financial.dto

import io.dustin.finstage.financial.domain.CashFlowStatement
import io.swagger.v3.oas.annotations.media.Schema
import java.time.LocalDate

data class CashFlowStatementResponse(
        @Schema(description = "기업 심볼", example = "AAPL")
        val symbol: String,

        @Schema(description = "보고일자", example = "2024-09-28")
        val date: LocalDate,

        @Schema(description = "영업활동 현금흐름", example = "118254000000")
        val operatingCashFlow: Long,

        @Schema(description = "투자활동 현금흐름", example = "2935000000")
        val investingCashFlow: Long,

        @Schema(description = "재무활동 현금흐름", example = "-121983000000")
        val financingCashFlow: Long,

        @Schema(description = "잉여현금흐름", example = "108807000000")
        val freeCashFlow: Long
) {
    companion object {
        fun from(cashFlow: CashFlowStatement): CashFlowStatementResponse {
            return CashFlowStatementResponse(
                    symbol = cashFlow.symbol,
                    date = cashFlow.date,
                    operatingCashFlow = cashFlow.operatingCashFlow,
                    investingCashFlow = cashFlow.investingCashFlow,
                    financingCashFlow = cashFlow.financingCashFlow,
                    freeCashFlow = cashFlow.freeCashFlow
            )
        }
    }
}
